.cirrus.yml

```diff
@@ -0,0 +1,54 @@
+check_android_task:
+  name: Run Android instrumented tests
+  env:
+    SCREENSHOTS_ROOT_DIR: /data/data/com.kaspersky.kaspressample/files/screenshots
+    TUTORIAL_SCREENSHOTS_ROOT_DIR: /data/data/com.kaspersky.kaspresso.tutorial/files/screenshots
+    ALLURE_SAMPLE_SCREENSHOTS_ROOT_DIR: /data/data/com.kaspersky.kaspresso.tutorial/files/screenshots
+    ALLURE_SAMPLE_FAIL_SCREENSHOTS_ROOT_DIR: /data/user/0/com.kaspersky.kaspresso.alluresupport.sample/files/screenshots
+  only_if: $CIRRUS_PR != '' && $CIRRUS_BASE_BRANCH == 'master'
+
+  container:
+    image: cirrusci/android-sdk:30
+    kvm: true
+    cpu: 6
+    memory: 24G
+  accept_licenses_script:
+    echo yes | sdkmanager --licenses
+  install_emulator_script:
+    sdkmanager --install "system-images;android-30;google_apis;x86"
+  create_avd_script:
+    echo no | avdmanager create avd --force
+    -n test
+    -k "system-images;android-30;google_apis;x86"
+  start_avd_background_script:
+    $ANDROID_HOME/emulator/emulator
+    -avd test
+    -no-audio
+    -no-window
+  assemble_instrumented_tests_script:
+    ./gradlew assembleDebugAndroidTest
+  wait_for_avd_script:
+    adb wait-for-device shell 'while [[ -z $(getprop sys.boot_completed) ]]; do sleep 3; done; input keyevent 82'
+  disable_animations_script: |
+    adb shell settings put global window_animation_scale 0.0
+    adb shell settings put global transition_animation_scale 0.0
+    adb shell settings put global animator_duration_scale 0.0
+  start_adb_server_background_script:
+    java -jar artifacts/adbserver-desktop.jar || true
+  run_tests_script: ./gradlew connectedDebugAndroidTest || mkdir screenshots && adb root && for i in {1..5}; do adb pull $SCREENSHOTS_ROOT_DIR sample_screenshots; \
```

Comment:
Please elaborate what is going on here? Why do we 5 times pull the `$SCREENSHOTS_ROOT_DIR` to different dirs on the host?

Reply:
After we do "adb root" connection is closed for a moment. So first attempt to pull screenshots usually fails